oak_remote_attestation/proto/v1/evidence.proto

```diff
@@ -0,0 +1,119 @@
+//
+// DICE provides a mechanism for combining software measurements and corresponding certificates into
+// a chain, where each element (called “layer”) represents a piece of software loaded into the VMs
+// memory. These layers are loaded sequentially, i.e. the previous layer loads the next layer. The
+// previous layer is also responsible for measuring the next layer, generating a private key and a
+// certificate for it.
+//
+// <https://trustedcomputinggroup.org/wp-content/uploads/TCG_DICE_Attestation_Architecture_r22_02dec2020.pdf>
+// <https://trustedcomputinggroup.org/wp-content/uploads/DICE-Layering-Architecture-r19_pub.pdf>
+
+enum TeePlatform {
+  UNSPECIFIED = 0;
+  AMD_SEV_SNP = 1;
+  INTEL_TDX = 2;
+}
+
+message RootLayerEvidence {
+  // The platform providing the attestation report.
+  TeePlatform platform = 0;
+  // TEE-specific attestation report acting as a non-standard certificate for the
+  // Layer0 ECA public key.
+  bytes remote_attestation_report = 1;
+  // Serialised ECA public key for Layer0.
+  bytes eca_public_key = 2;
+}
+
+message LayerEvidence {
+  string layer_name = 0;
+  // CBOR/COSE/CWT ECA certificate.
+  // <https://www.rfc-editor.org/rfc/rfc8392.html>
+  bytes eca_certificate = 1;
+}
+
+message ApplicationKeys {
+  // CBOR/COSE/CWT encryption public key certificate.
+  // <https://www.rfc-editor.org/rfc/rfc8392.html>
+  optional bytes encryption_public_key_certificate = 0;
+  // CBOR/COSE/CWT signing public key certificate.
+  // <https://www.rfc-editor.org/rfc/rfc8392.html>
+  optional bytes signing_public_key_certificate = 1;
+}
+
+message Evidence {
+  // Layer0 attestation evidence.
+  RootLayerEvidence root_layer_evidence = 0;
+  // Layer1..LayerN-1 attestation evidence.
+  repeated LayerEvidence layer_evidence = 1;
+  // Final layer keys.
+  optional ApplicationKeys application_keys = 2;
+}
+
+// Endorsement contains statements that some entity (e.g., a hardware provider) vouches for the
+// integrity of claims about the TEE or the software running on it.
+// The name is chosen to match the RATS terminology:
+// <https://www.rfc-editor.org/rfc/rfc9334.html#name-endorsements>
+message Endorsement {
+  // The serialized TEE certificate(s). The details of the format and how the certificate(s) are
+  // encoded into this byte array are implementation-specific. In case of AMD-SEV-SNP, as described
+  // in https://www.amd.com/system/files/TechDocs/57230.pdf, there are three different certificates
+  // packaged in two different files.
+  bytes tee_certificates = 1;
+
+  // Transparent Release endorsement.
+  TransparentReleaseEndorsement transparent_release_endorsement = 2;
+}
+
+message TransparentReleaseEndorsement {
+  // The serialized binary endorsement statement from the transparent release process.
+  bytes endorsement_statement = 1;
+  // The serialized Rekor LogEntry as proof of the inclusion of the endorsement statement in Rekor.
+  bytes rekor_log_entry = 2;
```

Comment:
I would the endorsement statement above to be the log entry, and this field to contain the signed inclusion proof.

Reply:
As far as I know, the Rekor log entry contains a signed timestamp which servers as the inclusion proof. So I don't see the distinction between log entry and a singed inclusion proof. Can you elaborate more?

Reply:
That is included in the log entry. The log entry ([example](https://github.com/project-oak/oak/blob/main/oak_remote_attestation_verification/testdata/logentry.json)) contains an inclusion proof filed, and a list of hashes. Does that make sense? 

Although, the current implementation of the verification does not verify the inclusion proof. Should it?

Reply:
Ah, I didn't know that it was included in the log entry.

I expected that we would verify inclusion in the log by checking the signed tree head and the Merkle tree, but thinking about it more, I guess it depends on the threat model. Verifying the full inclusion proof still relies on the signature of the tree head (since the client can be expected to verify the history of consistency proofs). I don't think there is a reason to trust the signature of the tree head more than the one that covers the body.

Perhaps verifying the inclusion proof can be done out of band by other verifiers that also check the consistency proofs between the various tree heads over time.

Reply:
> Perhaps verifying the inclusion proof can be done out of band by other verifiers that also check the consistency proofs between the various tree heads over time.

Yes. That is the reason the client does not verify the inclusion proof. We assume there are external verifiers for this, so the client only needs to verify the `SignedEntryTimestamp`, which is also the one verified in the [sigstore verify package](https://pkg.go.dev/github.com/sigstore/rekor/pkg/verify).